cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -2468,6 +2487,13 @@ def _get_datastores_for_profiles(self):
                                 custom_attributes[field] = v.value
                         datastores[summary.name][
                             "custom_attributes"] = custom_attributes
+
+                    if ('triggeredAlarmState' in objects and
+                            objects['triggeredAlarmState']):
+                        alarms = []
+                        for (_x, alarm) in objects['triggeredAlarmState']:
+                            alarms.append(alarm)
```

Comment:
I don't get this, yet. If I try to reproduce this via pyVmomi, I only get a list of `AlarmState` instances in `triggeredAlarmState`:
```(qa-de-1) [16] >>> ds.triggeredAlarmState
(qa-de-1) [16] <<< 
(vim.alarm.AlarmState) [
   (vim.alarm.AlarmState) {
      dynamicType = <unset>,
…
(qa-de-1) [17] >>> for (x, alarm) in ds.triggeredAlarmState:
              ...:     print(x)
              ...: 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-17-1f4fda3e6b2f> in <module>
----> 1 for (x, alarm) in ds.triggeredAlarmState:
      2     print(x)
      3 

TypeError: cannot unpack non-iterable vim.alarm.AlarmState object
```

Reply:
Ah. I get it. The `ArrayOf*` things we have a function in Nova to step over these iirc.. But makes sense. But then your `alarm` is really a list of `AlarmSate` objects and you just pick the first alarm wherever you use the saved "alarm". That doesn't sound right. We should rather iterate over `alarm` then and add each of these to the list of alarms.

Reply:
https://github.com/sapcc/nova/blob/8db25ac75ec0a993bb725800f8b4e1f9aedb409f/nova/virt/vmwareapi/vim_util.py#L143-L150

Reply:
I was just working on iterating over them instead of just using the first.   I'll have a new PR up in a minute